SetupTemplate/Product.wxs

```diff
@@ -0,0 +1,135 @@
+    <Directory Id="TARGETDIR" Name="SourceDir">
+      <Directory Id="ProgramFilesFolder">
+        <Directory Id="INSTALLFOLDER" Name="Cognite">
+          <Directory Id="ProductFolder" Name="$(var.product_short_name)">
+            <Directory Id="ConfigFolder" Name="config" />
+            <Directory Id="LogsFolder" Name="logs" />
+            <Directory Id="ExecutableFolder" Name="bin" />
+            <!-- Additional directories can be added here -->
+          </Directory>
+        </Directory>
+      </Directory>
+    </Directory>
+  </Fragment>
+
+  <Fragment>
+    <ComponentGroup Id="ExecutableComponentGroup">
+      <?if service='true' ?>
+        <Component Id="Primary_exe" Guid="*" Directory="ExecutableFolder">
+          <File Source="$(var.build_dir)\$(var.exe_name)" />
+
+          <?ifdef service_args?>
+            <ServiceInstall Id="ServiceInstaller"
+                            Type="ownProcess"
+                            Vital="yes"
+                            Name="$(var.product_short_name)"
+                            DisplayName="$(var.Name)"
+                            Description="$(var.Name) service."
+                            Start="auto"
+                            Account="LocalSystem"
+                            ErrorControl="ignore"
+                            Interactive="no"
+                            Arguments="$(var.service_args)">
+            
+              
+            
+              <util:ServiceConfig FirstFailureActionType="restart"
+                                  SecondFailureActionType="restart"
+                                  ThirdFailureActionType="restart"
+                                  RestartServiceDelayInSeconds="10" />
+              <ServiceDependency Id="Dnscache" />
+              <ServiceConfig DelayedAutoStart="yes" OnInstall="yes" OnReinstall="yes" />
+            </ServiceInstall>
+          <?else ?>
+            <ServiceInstall Id="ServiceInstaller"
+                        Type="ownProcess"
+                        Vital="yes"
+                        Name="$(var.product_short_name)"
+                        DisplayName="$(var.Name)"
+                        Description="$(var.Name) service."
+                        Start="auto"
+                        Account="LocalSystem"
+                        ErrorControl="ignore"
+                        Interactive="no">
+                <util:ServiceConfig FirstFailureActionType="restart"
+                                    SecondFailureActionType="restart"
+                                    ThirdFailureActionType="restart"
+                                    RestartServiceDelayInSeconds="10" />
+                <ServiceDependency Id="Dnscache" />
+                <ServiceConfig DelayedAutoStart="yes" OnInstall="yes" OnReinstall="yes" />
+              </ServiceInstall>
+            <?endif?>
+            <ServiceControl Id="StartService" Stop="both" Remove="uninstall" Name="OpcuaExtractor" Wait="no" />
```

Comment:
ExampleExtractor instead of OpcuaExtractor

Reply:
Yeah, this needs to be short-name. Good catch.